firstpartydata/first_party_data_test.go

```diff
@@ -0,0 +1,836 @@
+				userDataKey:        {{ID: "userDataId1", Name: "userDataName1"}},
+				appContentDataKey:  {{ID: "appDataId1", Name: "appDataName1"}},
+			},
+		},
+		{
+			description: "No Site, app or user data present",
+			input: []byte(`{
+  				"id": "bid_id",
+			 	"imp":[{"id":"impid"}]
+			}`),
+			output: []byte(`{
+  				"id": "bid_id",
+				"imp":[{"id":"impid"}]
+			}`),
+			expectedFpdData: map[string][]openrtb2.Data{
+				siteContentDataKey: nil,
+				userDataKey:        nil,
+				appContentDataKey:  nil,
+			},
+		},
+		{
+			description: "Site only data present",
+			input: []byte(`{
+  				"id": "bid_id",
+			 	"imp":[{"id":"impid"}],
+  				"site": {
+  				  "id": "reqSiteID",
+  				  "page": "test/page",
+    			  "content":{
+					"data":[
+						{ 
+						  "id": "siteDataId1",
+						  "name": "siteDataName1"
+						}
+					]
+				  }
+  				}
+			}`),
+			output: []byte(`{
+  				"id": "bid_id",
+				"imp":[{"id":"impid"}],
+				"site":{
+					 "id": "reqSiteID",
+  				     "page": "test/page",
+					 "content":{}
+				}
+			}`),
+			expectedFpdData: map[string][]openrtb2.Data{
+				siteContentDataKey: {{ID: "siteDataId1", Name: "siteDataName1"}},
+				userDataKey:        nil,
+				appContentDataKey:  nil,
+			},
+		},
+		{
+			description: "App only data present",
+			input: []byte(`{
+  				"id": "bid_id",
+			 	"imp":[{"id":"impid"}],
+  				"app": {
+  				  "id": "reqAppID",
+    			  "content":{
+					"data":[
+						{ 
+						  "id": "appDataId1",
+						  "name": "appDataName1"
+						}
+					]
+				  }
+  				}
+			}`),
+			output: []byte(`{
+  				"id": "bid_id",
+				"imp":[{"id":"impid"}],
+				"app":{
+					 "id": "reqAppID",
+					 "content":{}
+				}
+			}`),
+			expectedFpdData: map[string][]openrtb2.Data{
+				siteContentDataKey: nil,
+				userDataKey:        nil,
+				appContentDataKey:  {{ID: "appDataId1", Name: "appDataName1"}},
+			},
+		},
+		{
+			description: "User only data present",
+			input: []byte(`{
+  				"id": "bid_id",
+			 	"imp":[{"id":"impid"}],
+  				"site": {
+  				  "id":"reqSiteId"
+  				},
+  				"user": {
+  				  "id": "reqUserID",
+  				  "yob": 1982,
+  				  "gender": "M",
+				  "data":[
+						{ 
+						  "id": "userDataId1",
+						  "name": "userDataName1"
+						}
+					]
+  				},
+  				"app": {
+  				  "id": "appId"
+  				}
+			}`),
+			output: []byte(`{
+  				"id": "bid_id",
+				"imp":[{"id":"impid"}],
+  				"site": {
+  				  "id":"reqSiteId"
+  				},
+  				"user": {
+  				  "id": "reqUserID",
+  				  "yob": 1982,
+  				  "gender": "M"
+  				},
+  				"app": {
+  				  "id": "appId"
+  				}
+			}`),
+			expectedFpdData: map[string][]openrtb2.Data{
+				siteContentDataKey: nil,
+				userDataKey:        {{ID: "userDataId1", Name: "userDataName1"}},
+				appContentDataKey:  nil,
+			},
+		},
+	}
+	for _, test := range testCases {
+
+		var req openrtb2.BidRequest
+		err := json.Unmarshal(test.input, &req)
```

Comment:
Not sure why are we defining the input in these test cases to be JSON as we're just unmarshalling it here. Why not just define a `openrtb2.BidRequest` object directly and pass that in? that way you don't have to marshal the response either. You can just compare the objects directly.